database/migrations/2018_07_24_173448_create_categories_table.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateCategoriesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('categories', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name', 100 );
+            $table->unsignedInteger('parent_id')->default(0);
```

Comment:
chỗ này em để nullable và remove default(0) đi.
record nào có parent_id is null là category gốc